internal/terraform/resource_provider_mock_test.go

```diff
@@ -80,10 +82,19 @@ func simpleMockProvider() *MockProvider {
 	}
 }
 
-// ProviderSchema is a helper to convert from the internal GetProviderSchemaResponse to
-// a ProviderSchema.
-func (p *MockProvider) ProviderSchema() *ProviderSchema {
-	resp := p.getProviderSchema()
+// ProviderSchema is a helper to convert from the internal
+// GetProviderSchemaResponse to a ProviderSchema.
+func GetProviderSchema(p *testing_provider.MockProvider) *ProviderSchema {
```

Comment:
Can we make this an unexported function? And while we're doing this refactor, maybe we can also make `ProviderSchema` unexported?

Reply:
Done!